controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/SystemApplication.java

```diff
@@ -34,11 +37,16 @@ public NodeType nodeType() {
         return nodeType;
     }
 
-    /** Returns whether this system application has its own application package */
+    /** Returns whether this system application has an application package */
     public boolean hasApplicationPackage() {
         return nodeType == NodeType.proxy;
     }
 
+    /** Returns whether this system application must be upgraded in the declared order */
+    public boolean upgradeInOrder() {
```

Comment:
Why not let `all()` (maybe renamed) return `List<List<SystemApplcation>>` that will define both order (first list) and parallelism (second list) instead of having this method and enum order dictate that?

Reply:
This means that the first three applications will upgrade in parallel, which is a bit confusing, given that only the first two are mentioned here. It is also based on how the code is used. 

I'd prefer a DAG representation, and a topological sort when deploying, if this ever grows in complexity.